spec-tests/src/test/java/org/candlepin/spec/content/OwnerContentSpecTest.java

```diff
@@ -1157,4 +958,1038 @@ public void shouldPermitSafeParallelModification() throws Exception {
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedCids));
+        }
+
+        @Test
+        public void shouldAllowFilteringWithCustomExclusive() {
+            OwnerDTO owner = this.owners.get(1);
+
+            List<String> expectedCids = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .filter(ProductInfo::custom)
+                .flatMap(ProductInfo::content)
+                .map(ContentDTO::getId)
+                .toList();
+
+            List<ContentDTO> output = this.adminClient.ownerContent()
+                .getContentsByOwner(owner.getKey(), null, null, INCLUSION_INCLUDE, INCLUSION_EXCLUSIVE);
+
+            assertThat(output)
+                .isNotNull()
+                .map(ContentDTO::getId)
+                .containsAll(expectedCids)
+                .doesNotContainAnyElementsOf(this.getUnexpectedIds(expectedCids));
+        }
+
+        @Test
+        public void shouldErrorWithInvalidCustomInclusion() {
+            OwnerDTO owner = this.owners.get(1);
+
+            assertBadRequest(() -> this.adminClient.ownerContent()
+                .getContentsByOwner(owner.getKey(), null, null, INCLUSION_INCLUDE, "invalid_type"));
+        }
+
+        @Test
+        public void shouldAllowQueryingWithMultipleFilters() {
+            Random rand = new Random(14023);
+
+            OwnerDTO owner = this.owners.get(1);
+
+            // Hand pick a product to ensure that we have *something* that comes out of the filter, should
+            // our random selection process not have any intersection.
+            ProductInfo selected = this.productMap.values()
+                .stream()
+                .filter(this.buildOwnerProductPredicate(owner))
+                .filter(ProductInfo::active)
+                .filter(ProductInfo::custom)
+                .findAny()
+                .get();
+
+            List<String> pids = this.productMap.values()
+                .stream()
+                .filter(elem -> rand.nextBoolean())
+                .map(ProductInfo::id)
+                .collect(Collectors.toCollection(ArrayList::new));
+
+            List<String> cids = this.productMap.values()
+                .stream()
+                .filter(prod -> pids.contains(prod.id()))
+                .flatMap(ProductInfo::content)
+                .map(ContentDTO::getId)
+                .toList();
+
+            List<String> labels = this.productMap.values()
+                .stream()
+                .filter(elem -> rand.nextBoolean())
+                .flatMap(ProductInfo::content)
+                .filter(elem -> rand.nextBoolean())
+                .map(ContentDTO::getLabel)
+                .toList();
+
+            // Add all the product IDs from the select labels
+            pids.addAll(productMap.values()
+                .stream()
+                .filter(buildContentLabelPredicate(labels))
```

Comment:
The presence of this predicate suggests something has gone off the rails here.

Reply:
updated